platform_buckets_api/kube_client.py

```diff
@@ -111,11 +112,13 @@ def to_primitive(entry: PersistentCredentials) -> dict[str, Any]:
 class BucketCRDMapper:
     @staticmethod
     def from_primitive(payload: dict[str, Any]) -> BucketType:
+        owner = payload["metadata"]["labels"][OWNER_LABEL]
         common_kwargs = dict(
             id=payload["metadata"]["labels"][ID_LABEL],
             name=payload["metadata"]["labels"].get(BUCKET_NAME_LABEL),
-            owner=payload["metadata"]["labels"][OWNER_LABEL],
+            owner=owner,
             org_name=payload["metadata"]["labels"].get(ORG_NAME_LABEL),
+            project_name=payload["metadata"]["labels"].get(PROJECT_LABEL, owner),
```

Comment:
I think it cannot be done. Bucket credentials can span multiple projects and orgs within cluster because you specify bucket ids in POST credentials request. It's up to a user to decide upon buckets he wants to work with. Other project users may not have the same level of access.

Reply:
Agreed, thank you for the explanation!